code_challenges.rb

```diff
@@ -1,48 +1,92 @@
-# ASSESSMENT 4: Ruby Coding Practical Questions
 
-# --------------------1) Create a method that takes in an array and returns an array with only odd numbers sorted from least to greatest.
 
-fullArr1 = [4, 9, 0, '7', 8, true, 'hey', 7, 199, -9, false, 'hola']
-# Expected output: [-9, 7, 9, 199]
-fullArr2 = ['hello', 7, 23, -823, false, 78, nil, '67', 6, 'Number']
-# Expected output: [-823, 7, 23]
-
-
-
-
-
-# --------------------2) Create a class called Bike that is initialized with a model, wheels, and a frame size. The default number of wheels is 2. Create a get_info method that returns a sentence with all the data from the bike object.
-
-
-
-
-
-# Expected output example: 'The Trek bike has 2 wheels and a 168cm frame.'
-
-
-# -------------------3) Add a bell to the bike class and create a method that will ring the bell when the method is called.
-
-
-
-
-
-# Expected output example: my_bike.ring_bell => 'Beep beep!'
-
-
-# -------------------4) Add a speedometer to the Bike class. The speed should be initialized at 0.
-
-
-
-
-
-# Expected output example: my_bike.speed => 0
-
-
-# -------------------5) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed. The brake method should decrease the speed. The bike cannot go negative speeds.
-
-
-
-
-
-# Expected output example: my_bike.pedal_faster 10 => 10
-# Expected output example: my_bike.brake 15 => 0
+# # --------------------1) Create a method that takes in an array and returns an array with only odd numbers sorted from least to greatest.
+#
+full_arr_1 = [4, 9, 0, '7', 8, true, 'hey', 7, 199, -9, false, 'hola']
+# # Expected output: [-9, 7, 9, 199]
+full_arr_2 = ['hello', 7, 23, -823, false, 78, nil, '67', 6, 'Number']
+# # Expected output: [-823, 7, 23]
+new_arr = []
+
+full_arr_1.map do |value|
+  new_arr = full_arr_1.select { |element| element.is_a? (Integer)}
+  end
+
+puts new_arr.sort
+
+
```

Comment:
The logic is good, but there is still two issues: 1) It is not wrapped in a method. 2)The output is just numbers not an array.